tasks/main.yml

```diff
@@ -15,25 +11,69 @@
     - "{{ ansible_facts['distribution'] ~ '_' ~
           ansible_facts['distribution_version'] }}.yml"
   vars:
-    __template_vars_file: "{{ role_path }}/vars/{{ item }}"
-  when: __template_vars_file is file
+    __crypto_policies_vars_file: "{{ role_path }}/vars/{{ item }}"
+  when: __crypto_policies_vars_file is file
+
+- name: OS is supported
+  meta: end_host
+  when:
+    - not __crypto_policies_os_supported|bool
```

Comment:
This woill silently skip hosts where the role is not supported, right? I don't think this is ok. All other roles, IIRC, fail if the distribution is not supported.

Reply:
Right. I used sshd role for the inspiration here. Other, maybe better option might be using assert here so we fail if somebody accidentally runs this on unsupported os. Or is there better way to fail on unsupported hosts?

Reply:
We can discuss this topic later....

Reply:
Right - for now, just remove the use of `end_host` and assume there will be some other mechanism by which the user will either be able to skip this role on unsupported platforms, or otherwise handle failures gracefully.

In other system roles that have limited platform support (e.g. tlog), it is assumed that the user will structure their inventory such that hosts which do not support this functionality will be skipped.  For example, creating some sort of hostgroup e.g. `crypto_policies_hosts` so that only hosts in that group will use the role:
```yaml
- hosts: crypto_policies_hosts
  roles:
    - crypto_policies
  vars:
    crypto_policies_....
```

Reply:
Thank you for the comments. I that case, I think it does not make sense to keep `__crypto_policies_os_supported` variable at all so I will remove it. If somebody will try to run this on unsupported host, it will probably fail on the next task, when it will try to install packages.